gst/gst-kvs-plugin/src/GstPlugin.c

```diff
@@ -700,18 +659,18 @@ GstFlowReturn gst_kvs_plugin_handle_buffer(GstCollectPads* pads, GstCollectData*
             break;
     }
 
-    if (!IS_OFFLINE_STREAMING_MODE(pGstKvsPlugin->gstParams.streamingType)) {
-        if (pGstKvsPlugin->firstPts == GST_CLOCK_TIME_NONE) {
-            pGstKvsPlugin->firstPts = buf->pts;
-        }
-
-        if (pGstKvsPlugin->producerStartTime == GST_CLOCK_TIME_NONE) {
-            pGstKvsPlugin->producerStartTime = GETTIME() * DEFAULT_TIME_UNIT_IN_NANOS;
-        }
+    if (pGstKvsPlugin->firstPts == GST_CLOCK_TIME_NONE) {
+        pGstKvsPlugin->firstPts = buf->pts;
+    }
 
-        buf->pts += pGstKvsPlugin->producerStartTime - pGstKvsPlugin->firstPts;
+    if (pGstKvsPlugin->startTime == GST_CLOCK_TIME_NONE) {
+        pGstKvsPlugin->startTime = GETTIME() * DEFAULT_TIME_UNIT_IN_NANOS;
     }
 
+    buf->pts += pGstKvsPlugin->startTime - pGstKvsPlugin->firstPts;
+    buf->pts -= pGstKvsPlugin->firstPts;
```

Comment:
Is this extra line a mistake?

Reply:
I think so. I don't remember why it end up there, dont see it in original file either

Reply:
I tried to make sense of that math ... and checked the original file also ...